src/components/application-cost/index.js

```diff
@@ -0,0 +1,89 @@
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import AsyncResource from '../async-resource';
+import './style.css';
+import { getApplicationCost } from '../../state/application-cost';
+import applicationCostSet from '../../models/application-cost-set';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChartArea } from '@fortawesome/free-solid-svg-icons';
+import { CostContent } from './cost-content';
+import Spinner from '../spinner';
+import * as subscriptionCostApiActions from '../../state/subscriptions-cost-api/action-creators';
+import moment from 'moment';
+
+const periodDateFormat = 'YYYY-MM-DD';
+
+export const ApplicationCost = (props) => {
+  const { appName } = props;
+  const [applicationCostSet, setApplicationCostSet] = useState(null);
+  useEffect(() => {
+    props.subscriptionCostApiActions(props.appName, props.from, props.to);
+    setApplicationCostSet(props.applicationCostSet);
+    return () => {
+      props.unsubscriptionCostApiActions(props.appName, props.from, props.to);
```

Comment:
`unsubscriptionCostApiActions` is called each time `useEffect()` is called. This is hackish. Find a better way.

Reply:
We can try to use two useEffects - one for updating CostAllocationSet state, another -0 to subscribe-unsubscribe on change of appName and from/to

Reply:
Fixed with two `useEffect` sections

Reply:
LGTM. Maybe @keaaa can take a look to confirm that this pattern conforms to what we want to have moving forward.